homeassistant/components/template/binary_sensor.py

```diff
@@ -199,14 +217,27 @@ def _async_render(self):
                 return
             _LOGGER.error("Could not render template %s: %s", self._name, ex)
 
-        for property_name, template in (
-                ('_icon', self._icon_template),
-                ('_entity_picture', self._entity_picture_template)):
+        templates = dict((
+            ('_icon', self._icon_template),
+            ('_entity_picture', self._entity_picture_template),
+        ))
+
+        if self._attribute_templates is not None:
+            templates.update({'{}{}'.format(ATTRIBUTES_PREFIX, key): value
```

Comment:
This is a very complicated, please simplify. Do not set attributes on the Entity class. Instead, this should be something like:

```python
attrs = {}
for key, value in self._attribute_templates.items():
    try:
        attrs[key] = value.async_render()
    catch TemplateError as err:
        _LOGGER.error("Error rendering attribute %s: %s, key, err)
        
self._attributes = attrs
```

Reply:
I have now updated it to use that method (and added an additional test for this, since the extra exception handler pushed the coverage below the target).